src/components/ResourceList/ResourceList.tsx

```diff
@@ -791,29 +500,201 @@ class ResourceListInner extends React.Component<CombinedProps, State> {
+          value={sortValue}
+          disabled={selectMode}
+        />
+      </div>
+    ) : null;
 
-function defaultIdForItem(item: any, index: number) {
-  return Object.prototype.hasOwnProperty.call(item, 'id')
-    ? item.id
-    : index.toString();
-}
+  const alternateToolMarkup =
+    alternateTool && !sortingSelectMarkup ? (
+      <div className={styles.AlternateToolWrapper}>{alternateTool}</div>
+    ) : null;
 
-function isSmallScreen() {
-  return typeof window === 'undefined'
-    ? false
-    : window.innerWidth < SMALL_SCREEN_WIDTH;
-}
+  const headerTitleMarkup = (
+    <div className={styles.HeaderTitleWrapper} testID="headerTitleWrapper">
+      {headerTitle()}
+    </div>
+  );
+
+  const selectButtonMarkup = isSelectable ? (
+    <div className={styles.SelectButtonWrapper}>
+      <Button
+        disabled={selectMode}
+        icon={EnableSelectionMinor}
+        onClick={() => handleSelectMode(true)}
+      >
+        {i18n.translate('Polaris.ResourceList.selectButtonText')}
+      </Button>
+    </div>
+  ) : null;
+
+  const checkableButtonMarkup = isSelectable ? (
+    <div className={styles.CheckableButtonWrapper}>
+      <CheckableButton
+        accessibilityLabel={bulkActionsAccessibilityLabel()}
+        label={headerTitle()}
+        onToggleAll={handleToggleAll}
+        plain
+        disabled={loading}
+      />
+    </div>
+  ) : null;
+
+  const needsHeader =
+    isSelectable || (sortOptions && sortOptions.length > 0) || alternateTool;
+
+  const headerWrapperOverlay = loading ? (
+    <div className={styles['HeaderWrapper-overlay']} />
+  ) : null;
+
+  const showEmptyState = filterControl && !itemsExist && !loading;
+
+  const StickyHeader = (isSticky: boolean) => {
+    const headerClassName = classNames(
+      styles.HeaderWrapper,
+      sortOptions &&
+        sortOptions.length > 0 &&
+        !alternateTool &&
+        styles['HeaderWrapper-hasSort'],
+      alternateTool && styles['HeaderWrapper-hasAlternateTool'],
+      isSelectable && styles['HeaderWrapper-hasSelect'],
+      loading && styles['HeaderWrapper-disabled'],
+      isSelectable && selectMode && styles['HeaderWrapper-inSelectMode'],
+      isSticky && styles['HeaderWrapper-isSticky'],
+    );
+    return (
+      <div className={headerClassName} testID="ResourceList-Header">
+        <EventListener event="resize" handler={handleResize} />
+        {headerWrapperOverlay}
+        <div className={styles.HeaderContentWrapper}>
+          {headerTitleMarkup}
+          {checkableButtonMarkup}
+          {alternateToolMarkup}
+          {sortingSelectMarkup}
+          {selectButtonMarkup}
+        </div>
+        {bulkActionsMarkup}
+      </div>
+    );
+  };
+  const headerMarkup = !showEmptyState &&
+    (showHeader || needsHeader) &&
+    itemsExist && (
+      <div className={styles.HeaderOuterWrapper}>
+        <Sticky boundingElement={listRef.current}>
+          {StickyHeader(listRef.current !== null)}
```

Comment:
@amrocha @dleroux  Here is what I did to get the tests passing. Not sure I have context on the boolean being passed to StickyHeader. What is the thought there?

Reply:
I ended up doing a few things differently. Ill update in the main thread.